src/java/org/apache/cassandra/service/reads/repair/BlockingPartitionRepair.java

```diff
@@ -205,7 +199,9 @@ public void maybeSendAdditionalWrites(long timeout, TimeUnit timeoutUnit)
         if (awaitRepairsUntil(timeout + timeoutUnit.convert(mutationsSentTime, TimeUnit.NANOSECONDS), timeoutUnit))
             return;
 
-        EndpointsForToken newCandidates = writePlan.liveUncontacted();
+        EndpointsForToken newCandidates = writePlan.liveUncontacted().filter(
```

Comment:
can you rework this bit as:
`writePlan.consistencyLevel().isDatacenterLocal() ? writePlane.liveUncontacted().filter(InOutDcTester.replicas()) : writePlane.liveUncontacted();` so we're only filtering if we need to?